primary/tests/causal_completion_tests.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use std::time::Duration;
+use test_utils::cluster::Cluster;
+use tracing::{info, subscriber::set_global_default};
+use tracing_subscriber::filter::{EnvFilter, LevelFilter};
+
+#[ignore]
+#[tokio::test]
+async fn test_read_causal_signed_certificates() {
+    const CURRENT_ROUND_METRIC: &str = "narwhal_primary_current_round";
+
+    // Enabled debug tracing so we can easily observe the
+    // nodes logs.
+    setup_tracing();
+
+    let mut cluster = Cluster::new(None);
+
+    // start the cluster
+    let nodes = cluster.start(4).await;
+
+    // Let primaries advance little bit
+    tokio::time::sleep(Duration::from_secs(10)).await;
+
+    // Ensure all nodes advanced
+    for node in nodes {
+        let metric_family = node.registry.gather();
+
+        for metric in metric_family {
```

Comment:
Before this point, don't we need to add certificates to the nodes for processing? Or rather, I think it would be helpful for testing more specific scenarios, but don't quite know how to do that.

Reply:
The test introduced here may include useful scaffolding: https://github.com/MystenLabs/narwhal/pull/486